src/flamenco/snapshot/fd_snapshot_create.c

```diff
@@ -0,0 +1,1219 @@
+
+  FD_SCRATCH_SCOPE_BEGIN {
+
+  /* First convert the existing status cache into a snapshot-friendly format. */
+
+  fd_bank_slot_deltas_t slot_deltas_new = {0};
+  int err = fd_txncache_get_entries( snapshot_ctx->status_cache,
+                           &slot_deltas_new );
+  if( FD_UNLIKELY( err ) ) {
+    FD_LOG_WARNING(( "Failed to get entries from the status cache" ));
+    return -1;
+  }
+  ulong   bank_slot_deltas_sz = fd_bank_slot_deltas_size( &slot_deltas_new );
+  uchar * out_status_cache    = fd_valloc_malloc( snapshot_ctx->valloc,
+                                                  FD_BANK_SLOT_DELTAS_ALIGN, 
+                                                  bank_slot_deltas_sz );
+  fd_bincode_encode_ctx_t encode_status_cache = {
+    .data    = out_status_cache,
+    .dataend = out_status_cache + bank_slot_deltas_sz,
+  };
+  if( FD_UNLIKELY( fd_bank_slot_deltas_encode( &slot_deltas_new, &encode_status_cache ) ) ) {
+    FD_LOG_WARNING(( "Failed to encode the status cache" ));
+    return -1;
+  }
+
+  /* Now write out the encoded buffer to the tar archive. */
+
+  err = fd_tar_writer_new_file( snapshot_ctx->writer, FD_SNAPSHOT_STATUS_CACHE_FILE );
+  if( FD_UNLIKELY( err ) ) {
+    FD_LOG_WARNING(( "Failed to create the status cache file" ));
+    return -1;
+  }
+  err = fd_tar_writer_write_file_data( snapshot_ctx->writer, out_status_cache, bank_slot_deltas_sz );
+  if( FD_UNLIKELY( err ) ) {
+    FD_LOG_WARNING(( "Failed to create the status cache file" ));
+    return -1;
+  }
+  err = fd_tar_writer_fini_file( snapshot_ctx->writer );
+  if( FD_UNLIKELY( err ) ) {
+    FD_LOG_WARNING(( "Failed to create the status cache file" ));
+    return -1;
+  }
+
+  /* Registers all roots and unconstipates the status cache. */
+
+  fd_txncache_flush_constipated_slots( snapshot_ctx->status_cache );
+
+  fd_valloc_free( snapshot_ctx->valloc, out_status_cache );
+
+  return 0;
+
+  } FD_SCRATCH_SCOPE_END;
+
+}
+
+static inline int
+fd_snapshot_create_write_manifest_and_acc_vecs( fd_snapshot_ctx_t * snapshot_ctx,
+                                                fd_hash_t *         out_hash,
+                                                ulong *             out_capitalization ) {
+
+
+  fd_solana_manifest_serializable_t manifest = {0};
+  
+  /* Copy in all the fields of the bank. */
+
+  int err = fd_snapshot_create_populate_bank( snapshot_ctx, &manifest.bank );
+  if( FD_UNLIKELY( err ) ) {
+    FD_LOG_WARNING(( "Failed to populate the bank" ));
+    return -1;
+  }
+
+  /* Populate the rest of the manifest, except for the append vec index. */
+
+  manifest.lamports_per_signature                = snapshot_ctx->slot_bank.lamports_per_signature;
+  manifest.epoch_account_hash                    = &snapshot_ctx->slot_bank.epoch_account_hash;
+
+  /* TODO: The versioned epoch stakes needs to be implemented. */
+
+  manifest.versioned_epoch_stakes_len            = 0UL;
+  manifest.versioned_epoch_stakes                = NULL;
```

Comment:
(Just fyi:) not having this field will potentially break partitioned reward distribution if the snapshot is in one of the reward distribution blocks at the beginning of an epoch. See https://github.com/solana-foundation/solana-improvement-documents/pull/149

Reply:
For now, we could just not create snapshots whilst rewards are being distributed...

Reply:
Leaving a fixme, will do this in a follow up PR.